modules/service/src/main/scala/lucuma/odb/service/GeneratorParamsService.scala

```diff
@@ -78,24 +78,25 @@ object GeneratorParamsService {
       MissingData(none, paramName)
   }
 
-  def fromSession[F[_]: Sync](
-    session:  Session[F],
-    user:     User,
-    mService: ObservingModeServices[F]
-  ): GeneratorParamsService[F] =
-
+  def instantiate[F[_]: Concurrent](using Services[F]): GeneratorParamsService[F] =
     new GeneratorParamsService[F] {
 
       import lucuma.odb.sequence.gmos
 
+      override def select(
+        programId: Program.Id,
+        which:     Observation.Id
+      )(using Transaction[F]): F[Option[EitherNel[MissingData, GeneratorParams]]] =
+        selectAll(programId, List(which)).map(_.get(which))
```

Comment:
Was in necessary to move this down or is it just better esthetically?  Either way :+1: for sure, but am curious.